blockchain/blockchain_test.go

```diff
@@ -1006,7 +1029,7 @@ done:
 		case ev := <-chainSideCh:
 			block := ev.Block
 			if _, ok := expectedSideHashes[block.Hash()]; !ok {
-				t.Errorf("%d: didn't expect %x to be in side chain", i, block.Hash())
+				errs = append(errs, fmt.Errorf("%d: didn't expect %x to be in side chain", i, block.Hash()))
```

Comment:
I think we could stop testing here (`t.Errorf` -> `t.Fatalf`) and return the error. Then return type can be simply `err error`, not `errs []error`.